incubator/hnc/internal/reconcilers/object.go

```diff
@@ -359,6 +359,7 @@ func (r *ObjectReconciler) shouldSyncAsPropagated(log logr.Logger, inst *unstruc
 	// the type has 'Propagate' mode, the object will be overwritten.
 	mode := r.Forest.GetTypeSyncer(r.GVK).GetMode()
 	if mode == api.Propagate && srcInst != nil {
+		stats.OverwriteObject(r.GVK)
```

Comment:
There should be another place too, where a source is overwritten by "deletion" when the conflicting source in ancestor has `Remove` mode. WDYT @adrianludwin 

Something like this? https://github.com/kubernetes-sigs/multi-tenancy/blob/8da4e64a729160344d222f5bbc8e236e28d82c9a/incubator/hnc/internal/reconcilers/object.go#L394

Reply:
Good call, I missed that.

Reply:
@yiqigao217 sorry if i misunderstood you. `shouldSyncAsPropagated` checks if object is exists and on `propagate` mode. If `true` it will trigger metric collector AND call `syncPropagated` (which you suggest to put another collector here ) . Then i assume metrics will be called twice?

Reply:
@yiqigao217 I think that @frbimo is right, `shouldSyncAsPropagated` will never return true for conflicting source objects in `Remove` mode. So the situation you mentioned ("a source is overwritten by 'deletion'") shouldn't be able to occur, and the existing code is correct. Do you agree?

Reply:
Ah right. The conflicting source won't be deleted if the mode is from `Ignore` to `Remove` since the conflicting source doesn't have an `InheritedFrom` label. If the mode is first changed to `Propagate` and then `Remove`, it would be deleted but we record overwriting just once when it's overwritten in `Propagate` mode. LGTM.

Reply:
Basically we don't really have an delete "overwriting" at all.

Reply:
Agreed. We could try to "fix" that but that seems like a corner case of a corner case so I wouldn't worry about it too much myself. Thanks.